aptos-move/framework/supra-framework/sources/vesting_without_staking.spec.move

```diff
@@ -85,46 +85,60 @@ spec supra_framework::vesting_without_staking {
-    // }
-    //
-    // spec distribute_to_shareholder {
-    //     pragma verify = true;
-    //     // pragma opaque;
-    //     // modifies global<coin::CoinStore<SupraCoin>>(address_from);
-    //     let shareholder = shareholders_address[len(shareholders_address) - 1];
-    //     let shareholder_record = vesting_records[len(vesting_records) - 1];
-    //     let amount = min(shareholder_record.left_amount, fixed_point32::spec_multiply_u64(shareholder_record.init_amount, vesting_fraction));
-    //     let shareholder_amount = simple_map::spec_get(vesting_contract.shareholders, shareholder);
-    //     let post shareholder_amount_post = simple_map::spec_get(vesting_contract.shareholders, shareholder);
-    //     let address_from = signer::address_of(vesting_signer);
-    //     let address_to_beneficiary = simple_map::spec_get(vesting_contract.beneficiaries, shareholder);
-    //     let flag = simple_map::spec_contains_key(vesting_contract.beneficiaries, shareholder);
-    //     // Ensure that the amount is transferred to the beneficiary and the amount is substract from left_amount if the beneficiary exists
-    //     ensures (flag && address_from != address_to_beneficiary)
-    //         ==> (coin::balance<SupraCoin>(address_to_beneficiary) == old(coin::balance<SupraCoin>(address_to_beneficiary)) + amount
-    //             && coin::balance<SupraCoin>(address_from) == old(coin::balance<SupraCoin>(address_from)) - amount
-    //             && shareholder_amount_post.left_amount == shareholder_amount.left_amount - amount);
-    //     // Ensure that the amount is transferred to the shareholder and the amount is substract from left_amount if the beneficiary doesn't exist
-    //     ensures (!flag && address_from != shareholder)
-    //         ==> (coin::balance<SupraCoin>(shareholder) == old(coin::balance<SupraCoin>(shareholder)) + amount
-    //             && coin::balance<SupraCoin>(address_from) == old(coin::balance<SupraCoin>(address_from)) - amount
-    //             && shareholder_amount_post.left_amount == shareholder_amount.left_amount - amount);
-    //     // Ensure the length of the shareholders_address and vesting_records are the same if they are the same before the function call
-    //     ensures (len(old(shareholders_address)) == len(old(vesting_records))) ==> (len(shareholders_address) == len(vesting_records));
-    // }
+    spec vest_individual {
+        pragma verify = true;
+        pragma aborts_if_is_partial = true;
+        include VestingContractActive;
+        let vesting_contract_pre = global<VestingContract>(contract_address);
+        let post vesting_contract_post = global<VestingContract>(contract_address);
+        let vesting_schedule = vesting_contract_pre.vesting_schedule;
+        let last_vested_period = vesting_schedule.last_vested_period;
+        let next_period_to_vest = last_vested_period + 1;
```

Comment:
Again, since `last_vested_period` is maintained on per shareholder basis, this doesn't make sense.

Reply:
I think this addition at spec level is just store the value of `last_vested_period` + 1. why this is needed? because this is for write the spec for pre and post state stays the same.